lib/AST/TypeSubstitution.cpp

```diff
@@ -1056,7 +1057,8 @@ static bool canSubstituteTypeInto(Type ty, const DeclContext *dc,
 
   case OpaqueSubstitutionKind::SubstituteNonResilientModule:
     // Can't access types that are not public from a different module.
-    if (dc->getParentModule() == typeDecl->getDeclContext()->getParentModule())
+    if (dc->getParentModule() == typeDecl->getDeclContext()->getParentModule() &&
```

Comment:
I think this condition is the same as `if (typeDecl->isResilient(contextExpansion))` no?

Reply:
I don't believe so:

We arrive at this if statement with the assumption `SubstituteNonResilientModule` (the naming decl of the opaque type is in a non-resilient module).

This means  the`typeDecl->isResilient(dc->getParentModule(), contextExpansion)` code would return `false` if `dc->getParentModule() == typeDecl->getDeclContext()->getParentModule()` and `contextExpansion == ResilienceExpansion::Maximal`.

```
bool NominalTypeDecl::isResilient() const {                                     
  if (!isFormallyResilient())                                                   
    return false;                                                               
                                                                                
  return getModuleContext()->isResilient(); // would also return false but we don't get here
} 
bool NominalTypeDecl::isResilient(ModuleDecl *M,                                
                                  ResilienceExpansion expansion) const {        
  switch (expansion) {                                                          
  case ResilienceExpansion::Minimal:                                            
    return isResilient();                                                        
  case ResilienceExpansion::Maximal:                                            
    // We can access declarations from the same module                          
    // non-resiliently in a maximal context.                                             
    if (M == getModuleContext()) {                                              
      return false;                    // <<< === will return false                                                  
    }
    ...                                                                
    // Otherwise, we have to access the declaration resiliently if it's         
    // resilient anywhere.                                                      
    return isResilient();                                                       
  } 
```